### PR TITLE
[AGENT] Add default-on prompt vs no-prompt transcription voting

### LINKED ISSUE #114: [AGENT] Vote transcription with arbiter
[AGENT]

## Summary
Add an optional vote transcription path that runs multiple transcriptions on the same snippet (prompt and no prompt or fast and slow) and selects the best via an arbiter.

## Motivation
The audit shows prompt based and no prompt transcriptions can diverge on short, repetitive clips. A vote path can reduce runaway repetitions and improve correctness without changing base behavior.

## Proposal
- Run two transcriptions for select snippets: prompt and no prompt, or fast and slow.
- Add an arbiter that picks the best result using logprob metrics, repetition penalties, and basic text quality checks.
- Gate the vote path to low confidence cases only, for example prompt echo detected, low logprob confidence, or large divergence between fast and slow.

## Open questions
- Which gating criteria provide the best cost quality tradeoff?
- Should the arbiter be deterministic rules or a small model call?

### PR DESCRIPTION
**Pull request by BASIC-BIT**
[AGENT] ## Summary
- add a dual-path vote flow in transcription that runs prompt and no-prompt candidates only for low-confidence slow-pass snippets
- select the safer candidate via deterministic scoring (suppression, prompt-echo/rate flags, quiet-audio, and logprob metrics) to reduce hallucinated prompt carryover without adding an extra arbiter model call
- enable the feature by default via runtime config (`transcription.vote.enabled=true`), wire it through config/registry/runtime resolution, and document the new guardrail

[AGENT] ## Testing
- yarn lint:check
- yarn prettier:check
- yarn build:all
- yarn test
- yarn test:e2e
- yarn checkov
- yarn prompts:check
- yarn llm-connections:check
- yarn code:stats

[AGENT] Closes #114

### CODE REVIEW

File: test/utils/transcriptionVote.test.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    ).toBe(true);
+    expect(
+      hasLowConfidenceLogprobs({
+        avgLogprob: -0.4,
+        minLogprob: -1,
+        tokenCount: 4,
+      }),
+    ).toBe(false);
+  });
+
```
[the remark this answers is not in the document]
Comment (BASIC-BIT) [reply test/utils/transcriptionVote.test.ts]: [AGENT] Added this test in `test/utils/transcriptionVote.test.ts` to explicitly cover `hasLowConfidenceLogprobs(undefined) === false` so the undefined-metrics edge case is locked in.

File: test/utils/transcriptionVote.test.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        hasPrompt: false,
+        passMode: "slow",
+        primaryCandidate,
+      }),
+    ).toEqual({
+      shouldRun: false,
+      reasons: [],
+    });
+  });
+
```
[the remark this answers is not in the document]
Comment (BASIC-BIT) [reply test/utils/transcriptionVote.test.ts]: [AGENT] Added this coverage in `test/utils/transcriptionVote.test.ts` with `enabled: false` to verify the vote gate remains off even when other trigger signals are present.
